java/code/src/com/redhat/rhn/domain/errata/ErrataFactory.java

```diff
@@ -972,6 +998,25 @@ public static void syncErrataDetails(ClonedErrata cloned) {
             Bug cloneB = createBug(bugIn.getId(), bugIn.getSummary(), bugIn.getUrl());
             cloned.addBug(cloneB);
         }
+
+        if (previousAdvisoryStatus != cloned.getAdvisoryStatus()) {
+            boolean retract = (cloned.getAdvisoryStatus() == AdvisoryStatus.RETRACTED);
+            cloned.getChannels().forEach(c -> {
+                processRetracted(cloned.getId(), c.getId(), retract);
+                ChannelFactory.refreshNewestPackageCache(c, "sync errata");
+            });
```

Comment:
Good point! Then the errata-packages-channel combo will get inserted to the cache, which shouldn't hurt. But I think a similar test as the one you've commented above should be done here.

Reply:
I've adjusted the code, so that the cache update (either removing or updating) only happens, when exactly one patches is retracted (corresponds to the "retracting a patch" and "unretracting a patch" scenarios).